src/core/CustomData.cpp

```diff
@@ -53,20 +66,28 @@ bool CustomData::contains(const QString& key) const
 
 bool CustomData::containsValue(const QString& value) const
 {
-    return asConst(m_data).values().contains(value);
+    for (auto i = m_data.constBegin(); i != m_data.constEnd(); ++i) {
+        if (i.value().value == value) {
+            return true;
+        }
+    }
+    return false;
 }
 
-void CustomData::set(const QString& key, const QString& value)
+void CustomData::set(const QString& key, CustomDataItem item)
 {
     bool addAttribute = !m_data.contains(key);
-    bool changeValue = !addAttribute && (m_data.value(key) != value);
+    bool changeValue = !addAttribute && (m_data.value(key).value != item.value);
 
     if (addAttribute) {
         emit aboutToBeAdded(key);
     }
 
+    if (!item.lastModified.isValid()) {
+        item.lastModified = Clock::currentDateTimeUtc();
```

Comment:
Shouldn't we set this to the current time no matter what, especially if it is valid? We do that in the `rename` function.

Reply:
No, this will only fill in a date when no date has been set in the file. We don't want to overwrite dates from the file, because that would make this field useless, because that would be like changing the modification time of a file on read.

Reply:
Sorry I missed that item was passed in to this function, not a simple value.